### ISSUE TITLE
compile=true ignores already built assets

### ISSUE THREAD

**Issue by dapriett** [adunkman/connect-assets#322]
From reading the docs, it says

```
compile	true	Should assets be compiled if they don’t already exist in the buildDir?
```

But it looks like assets are never being read from the manifest if compile is not false (https://github.com/adunkman/connect-assets/blob/master/lib/assets.js#L213), ignoring already precompiled files.

Should it be doing something like this so that it checks the manifest first, before it attempts to compile?

```
Assets.prototype.getAssetByPath = function (pathname, options) {
			var asset;
			if (this.manifest) {
				var assetTitle = this.manifest.assets[pathname];
				asset = this.manifest.files[assetTitle];
			}
			if (!asset && this.options.compile) return this.environment.findAsset(pathname, options);
			return asset;
		};
```

**Comment by thawatchai** [comment]
:+1: I have the same problem.

**Comment by blakevanlan** [comment]
Fixed! Thanks for the PR @dapriett. Published in v5.2.0.
